### PR TITLE
Add GH Actions for unit and integration tests

### PR DESCRIPTION
**Pull request by chris-hinds**
Resolves #NUMBER

**Overall change:**
- Adds a GH Action for running unit tests
- Adds a GH Action for running integration tests
- Adds a GH Action for running E2E tests with Cypress
- Adds a GH Action for cancelling previously running builds if a new one starts
- Adds a GH Action for running, licence checker, dependencies checker and chromatic QA (Chromatic QA is disabled now though as CodeBuild runs this for now)

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: .github/workflows/simorgh-misc-checks.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: Simorgh CI - Miscellaneous Checks
```
Comment (karinathomasbbc) [inline .github/workflows/simorgh-misc-checks.yml]: Mind if we give this more descriptive name e.g. Licences, Dependencies, Lint, Chromatic UI (I know it's a mouthful - it's just easier to see on the PR)

**Review by karinathomasbbc — APPROVED** [review]

File: .github/workflows/simorgh-e2e-tests.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: Simorgh CI - E2E Tests
```
Comment (karinathomasbbc) [inline .github/workflows/simorgh-e2e-tests.yml:1]: Should we keep Simorgh CI, or just rename to `Cypress End-to-End Tests`

File: .github/workflows/simorgh-integration-tests.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+name: Simorgh CI - Integration Tests
```
Comment (karinathomasbbc) [inline .github/workflows/simorgh-integration-tests.yml:1]: Should we keep Simorgh CI in the description?

**Comment by karinathomasbbc** [comment]
I have changed my mind - keep Simorgh CI! 🍎

File: .github/workflows/simorgh-integration-tests.yml (hunk 3 above)
Comment (chris-hinds) [reply .github/workflows/simorgh-integration-tests.yml:1]: we don't have too but I feel it gives this actions a nice separation from other checks we have running

Comment (karinathomasbbc) [reply .github/workflows/simorgh-integration-tests.yml:1]: Yes, it looks better when they are namespaced with Simorgh
image

File: .github/workflows/simorgh-misc-checks.yml (hunk 1 above)
Comment (chris-hinds) [reply .github/workflows/simorgh-misc-checks.yml]: Updated

**Review by rhenshaw56 — APPROVED** [review]

**Pull request merged**